### DISCUSSION TITLE
Digitally sign document with X509 certificate

### DISCUSSION THREAD

**Discussion by bjorn-malmo** [QuestPDF/QuestPDF#733] [Ideas]
Thank you for a great library!

We would really appreciate the possibility to digitally sign the created PDF-file using an X509 certificate. This would improve future document security vastly.
Is the pdf standard documentation for adding digital signatures public?

Regards Björn

**Comment by MarcinZiabek** [comment]
Hi 😀

The functionality of signing PDF files with certificates is currently being considered for future releases, but it is of lower priority compared to ongoing efforts.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Digital signature support is tracked in [Issue #1193](https://github.com/QuestPDF/QuestPDF/issues/1193) and remains in our backlog. In the meantime, a common workaround is to generate the PDF with QuestPDF and then sign it using a dedicated signing library such as iText.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
